scripts/pipeline/orthology_benchmark.py

```diff
@@ -544,47 +542,41 @@ def calculate_goc_genes(gene1: str, gene2: str, neighbourhood1: pandas.DataFrame
     strand2 = neighbourhood2.iloc[index2]["strand"]
 
     if strand1 != strand2:
+        # Reverse `neighbourhood2` to potentially speed up search for putative orthologous matches later
         neighbourhood2 = neighbourhood2.reindex(index=neighbourhood2.index[::-1])
         neighbourhood2.reset_index(inplace=True)
         neighbourhood2.drop("index", 1, inplace=True)
+        index2 = neighbourhood2.index[neighbourhood2["gene_id"] == gene2][0]
         strand_mismatch = True
     else:
         strand_mismatch = False
 
+    # Remove `gene1` and `gene2` from `neighbourhood1` and `neighbourhood2` before looking for putative
+    # orthologs between genes in `neighbourhood1` and `neighbourhood2`
+    neighbourhood1.drop(index=index1, inplace=True)
+    neighbourhood1.reset_index(inplace=True)
+    neighbourhood1.drop("index", 1, inplace=True)
+
+    neighbourhood2.drop(index=index2, inplace=True)
+    neighbourhood2.reset_index(inplace=True)
+    neighbourhood2.drop("index", 1, inplace=True)
+
     shared_orthologs = 0
-    prev_match = -1
     for i in neighbourhood1.iterrows():
         gene_i = i[1]["gene_id"]
-
-        if gene_i == gene1:  # Skip gene1
-            continue
-
         strand_i = i[1]["strand"]
 
-        # Check if there are orthologous matches starting from (`prev_match` + 1)th position in
-        # `neighbourhood2`
-        for index, row in itertools.islice(neighbourhood2.iterrows(), prev_match + 1, None):
-            gene_j = row["gene_id"]
-
-            if gene_j == gene2:
-                prev_match = index  # Current index
-                continue  # Skip gene2
+        # Check if there are orthologous matches in `neighbourhood2`
+        for j in neighbourhood2.iterrows():
+            gene_j = j[1]["gene_id"]
 
-            # Check that gene_i and gene_j share directionality; if not, go to next gene in `neighbourhood2`
-            strand_j = row["strand"]
+            # Check the directionality of gene_i and gene_j
+            strand_j = j[1]["strand"]
             if (strand_mismatch and strand_i == strand_j) or (not strand_mismatch and strand_i != strand_j):
                 continue
 
             if (gene_i, gene_j) in orthologs or (gene_j, gene_i) in orthologs:
-                # If prev_match >=0 and gap too big, i.e. abs(j-prev_match) > allowed_gap + 1:
-                #   Ignore that match
-                # Else:
-                #   shared_orthologs++
-                curr_pos_j = index
-                if prev_match < 0 or abs(curr_pos_j - prev_match) <= allowed_gap + 1:
-                    shared_orthologs += 1
-
-                prev_match = curr_pos_j
+                shared_orthologs += 1
                 break  # Don't look for multiple matches
 
     return (shared_orthologs / (2 * n)) * 100
```

Comment:
We were talking about keeping it simple and dividing by `2*n` which is the maximum number of neighbours fetched for `gene1`. But `gene1` can have fewer neighbours due to the fragmented assembly or simply by being the first or last one on the chromosome. Based on my code it is easy to divide by the number of neighbours we actually have for `gene1` but that has its pros and cons.

Reply:
@dthybert Not sure if "👍" means to keep it as it is or divide by the actual number of neighbours?